tests/end-to-end.rs

```diff
@@ -1,107 +1,735 @@
+                    "Stack.cs",
+                    "Program.cs",
+                ),
+                "golang" => (
+                    Box::new(cdk_app_code_writers::Golang {}) as Box<dyn CdkAppCodeWriter>,
+                    Box::<Golang>::default() as Box<dyn Synthesizer>,
+                    "stack.go",
+                    "app.go",
+                ),
+                "java" => (
+                    Box::new(cdk_app_code_writers::Java {}) as Box<dyn CdkAppCodeWriter>,
+                    Box::<Java>::default() as Box<dyn Synthesizer>,
+                    "src/main/java/com/myorg/Stack.java",
+                    "src/main/java/com/myorg/MyApp.java",
+                ),
+                "python" => (
+                    Box::new(cdk_app_code_writers::Python {}) as Box<dyn CdkAppCodeWriter>,
+                    Box::new(Python {}) as Box<dyn Synthesizer>,
+                    "stack.py",
+                    "app.py",
+                ),
+                "typescript" => (
+                    Box::new(cdk_app_code_writers::Typescript {}) as Box<dyn CdkAppCodeWriter>,
+                    Box::new(synthesizer::Typescript {}) as Box<dyn Synthesizer>,
+                    "stack.ts",
+                    "app.ts",
+                ),
+                other => panic!("Unsupported language: {other}"),
+            };
+
+        let cursor: Cursor<&[u8]> =
+            std::io::Cursor::new(include_bytes!("./end-to-end-test-snapshots.zip"));
+        let snapshots_zip = zip::read::ZipArchive::new(cursor)
+            .expect("Failed to convert end-to-end-test-snapshots.zip contents into ZipArchive");
+
+        EndToEndTest {
+            name,
+            lang,
+            original_template,
+            stack_name,
+            cdk_app_code_writer,
+            cdk_from_cfn_synthesizer,
+            cdk_stack_filename,
+            cdk_app_filename,
+            language_boilerplate_dir: String::from(format!(
+                "tests/end-to-end/app-boilerplate-files/{lang}"
+            )),
+            test_working_dir: String::from(format!("tests/end-to-end/{name}-{lang}-working-dir")),
+            expected_outputs_dir: String::from(format!("{name}/{lang}")),
+            snapshots_zip,
+            skip_cdk_synth: skip_cdk_synth.contains(&lang),
+        }
+    }
+
+    fn run(&mut self) {
+        // GIVEN
+        self.create_stack();
+
+        // WHEN
+        let cdk_stack_definition = self.run_cdk_from_cfn();
+
+        // THEN
+        self.check_cdk_stack_def_matches_expected(&cdk_stack_definition);
+
+        if self.skip_cdk_synth {
+            self.update_cdk_stack_def_snapshot(&cdk_stack_definition);
+        } else {
+            self.synth_cdk_app(&cdk_stack_definition);
+
+            self.diff_original_template_with_new_templates();
+
+            self.check_new_templates_and_diffs_match_expected();
+
+            self.update_snapshots(&cdk_stack_definition);
```

Comment:
check_new_templates... and update_snapshots both exit immediately depending on the value `UPDATE_SNAPSHOTS`. Do we need the early return within those functions or can we pull those checks and never call them to begin with? It would make the control flow a bit easier to follow.

Reply:
yes, I will make that change if it's easier to read that way!